### PR TITLE
add nodepool governance capability proposal

### PR DESCRIPTION
**Pull request by Peeknut**
/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rambohe-ch** [comment]
@Peeknut please fix the markdown check errors.

**Comment by rambohe-ch** [comment]
@Peeknut according to the name votes in issue #786 , `pool-coordinator` will be used as name for the new component name. please rename `pool-spirit` as `pool-coordinator` in this proposal.

**Comment by Peeknut** [comment]
> @Peeknut according to the name votes in issue #786 , `pool-coordinator` will be used as name for the new component name. please rename `pool-spirit` as `pool-coordinator` in this proposal.

down

File: docs/proposals/20220307-nodepool-governance-capability.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+deploy the pool-coordinator when the number of nodes in the NodePool is less than 3, or if a pool-coordinator has been deployed in the NodePool.
+2. When pool-coordinator is scheduled, coordinator-controller prepares the tls server certificate for pool-coordinator,
+saves the certificate in secret and mounts it to pool-coordinator. Note that the tls server certificate for each pool-coordinator is different
+because certificate includes the pool-coordinator service clusterIP and the node IP.
+3. coordinator-controller generates kubeconfig for users to access pool-coordinator. The server address in kubeconfig is set to
+https://{nodeIP}:10270. In addition, the client certificate authority in kubeconfig should be restricted. For details,
+please refer to the [kubeconfig of pool-coordinator](#kubeconfig-for-users-to-access-pool-coordinator).
+4. When the pool-coordinator is rebuilt, coordinator-controller will clean up and rebuild the tls server certificate.
+5. When Spriit CR is deleted, coordinator-controller will delete the NodePool information in YurtAppSet. It also cleans up
```
Comment (rambohe-ch) [inline docs/proposals/20220307-nodepool-governance-capability.md:143]: Spriit --> PoolCoordinator

File: docs/proposals/20220307-nodepool-governance-capability.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+### Principles
+
+- No modification to k8s, keeping it non-invasive.
+- The `NodePool governance capability` has no impact on the OpenYurt, and can be free to go online and offline.
+- The cloud center is the source of truth to ensure the consistency of cloud and edge data.
+- The design is simple, reliable and stable.
+
+### Architecture
+
+To provide `NodePool governance capability`, we add a component to the NodePool, tentatively named `pool-coordinator`.
```
Comment (rambohe-ch) [inline docs/proposals/20220307-nodepool-governance-capability.md:96]: ~~tentatively~~

File: docs/proposals/20220307-nodepool-governance-capability.md (hunk 1 above)
Comment (Peeknut) [reply docs/proposals/20220307-nodepool-governance-capability.md]: down

File: docs/proposals/20220307-nodepool-governance-capability.md (hunk 2 above)
Comment (Peeknut) [reply docs/proposals/20220307-nodepool-governance-capability.md]: down

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**